Added BooleanField field Type

rowan08 opened the pull request:
Added the field to Field.h, Model.h and Model.cpp.
Also added to Model::insertBatch() to insert data for BOOL field types
Tested by adding a BooleanField, proteinIsDrugTarget, to test.cpp and setting the value to true. The script runs fine.
Issue: https://github.com/davidbrownza/CORMAP/issues/9

rowan08 committed:
Added BooleanField field Type
Added the field to Field.h, Model.h and Model.cpp.
Also added to Model::insertBatch() to insert data for BOOL field types
Tested by adding a BooleanField, proteinIsDrugTarget, to test.cpp and setting the value to true. Script runs fine.
Issue: https://github.com/davidbrownza/CORMAP/issues/9

davidbrownza commented on Model.cpp:
```diff
@@ -129,6 +137,15 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
                         conn.setString(paramNum, i->getValue());
                     }
                 }
+                else if(f->getType() == BOOLEAN)
```
Just a side note, we should probably convert this if statement to a switch at some point in the future. Long term, there is a repeating pattern here that we could probably refactor.

davidbrownza replied:
For a start, the null check could be done before this if statement, which would mean we wouldn't have to repeat it within each block.

davidbrownza replied:
Actually, now is as good a time as any to do this. Convert this whole if statement to a switch and perform the null check before the switch.

davidbrownza reviewed (changes requested):
You did everything right, but I have added on an extra requirement :P

rowan08 committed:
Changed field-type check in Model.cpp from if-statements to a switch statement.
Null check is also now done before checking field-type.
Did not alter test scripts. Compiles and runs fine.
Issue: https://github.com/davidbrownza/CORMAP/issues/9

davidbrownza commented on Model.cpp:
```diff
@@ -92,41 +100,46 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
+                    switch(f->getType()){
+
+                        case INTEGER: 
+                        {
```
Remove these brackets in the cases, they are not necessary.

rowan08 replied:
If I remove the brackets (and nothing else), it throws errors along the lines of `error: jump to case label [-fpermissive]`,` error:   crosses initialization of ‘IntegerField* i’`  and ` error: conflicting declaration ‘BooleanField* i’ `.

This seems to have something to do with declaring variables within a switch statement, which is not allowed for reasons I don't entirely follow (see below...), but they recommend adding an additional layer of scope to fix this problem. 

https://stackoverflow.com/questions/449736/why-can-t-variables-be-declared-in-a-switch-statement

davidbrownza replied:
Does it work if you move the `break` inside the brackets:
```
case INTEGER:
{
    IntegerField * i = static_cast <IntegerField*>(f);
    conn.setInt(paramNum, i->getValue());
    break;
}
```
If not, it is fine, just fix up the indentation like so:
```
case INTEGER:
    {
        IntegerField * i = static_cast <IntegerField*>(f);
        conn.setInt(paramNum, i->getValue());
    }
    break;
```

davidbrownza commented on Model.cpp:
```diff
@@ -92,41 +100,46 @@ int Model::insertBatch(vector<Model*> models, int batchsize, Mode mode)
-                {
+                if (f->isNull()) {
+                    conn.setNull(paramNum);
+                } else {
```
You can remove the `else` block if you use a `continue` statement in the `if` block after setting the parameter to null.

davidbrownza reviewed (changes requested):
Two small changes.

rowan08 committed:
Added continue-statement after checking for null value during insert (instead of using else)
Changed layout of cases so break-statements are within the curly braces
Testing scripts unchanged. Ran fine.
Issue: https://github.com/davidbrownza/CORMAP/issues/9
